lib/src/components/NxFilterInput/__tests__/NxFilterInput.test.tsx

```diff
@@ -5,160 +5,109 @@
-import { PrivateNxTextInput } from '../../NxTextInput/NxTextInput';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
-import NxButton from '../../NxButton/NxButton';
-import Close from '../../../icons/Close';
 
 describe('NxFilterInput', function() {
   const minimalProps = { value: '' },
-      shallowComponent = enzymeUtils.getShallowComponent<Props>(NxFilterInput, minimalProps),
-      mountedComponent = enzymeUtils.getMountedComponent<Props>(NxFilterInput, minimalProps);
+      quickRender = rtlRender<Props>(NxFilterInput, minimalProps),
+      renderEl = rtlRenderElement<Props>(NxFilterInput, minimalProps);
 
-  it('renders a PrivateNxTextInput with the nx-filter-input class', function() {
-    const component = shallowComponent();
-    expect(component).toHaveClassName('nx-filter-input');
-    expect(component).toMatchSelector(PrivateNxTextInput);
+  it('renders an Input', function() {
+    expect(quickRender().getByRole('textbox').tagName).toBe('INPUT');
   });
 
-  it('adds custom classnames as specified', function() {
-    const component = shallowComponent({ className: 'foo' });
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-filter-input');
+  it('adds additional specified classNames to the top-level element', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
+
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('passes misc props to the PrivateNxTextInput', function() {
-    const onChange = jest.fn(),
-        onKeyPress = jest.fn(),
-        component = shallowComponent({
-          value: 'foo',
-          onChange,
-          onKeyPress,
-          id: 'bar'
-        });
-
-    expect(component).toHaveProp('value', 'foo');
-    expect(component).toHaveProp('onChange', onChange);
-    expect(component).toHaveProp('onKeyPress', onKeyPress);
-    expect(component).toHaveProp('id', 'bar');
+  it('passes additional props to the Input', function() {
+    const input = quickRender({ id: 'foo', lang: 'en-US' }).getByRole('textbox');
+
+    expect(input).toHaveAttribute('id', 'foo');
+    expect(input).toHaveAttribute('lang', 'en-US');
   });
 
-  it('does not pass validatable, validationErrors, or type props to the PrivateNxTextInput', function() {
-    const component = shallowComponent({
+  it('does not pass validatable, validationErrors, or type props to the Input', function() {
+    const input = quickRender({
       validatable: true,
       validationErrors: 'It\'s all wrong',
       type: 'textarea'
-    } as Partial<Props>);
-
-    expect(component).not.toHaveProp('validatable');
-    expect(component).not.toHaveProp('validationErrors');
-    expect(component).not.toHaveProp('type');
-  });
+    } as Partial<Props>).getByRole('textbox');
 
-  it('passes isPristine = false to the PrivateNxTextInput', function() {
```

Comment:
The `isPristine=false` is passed down to `PrivateNxTextInput` and is used to check whether to have `showValidationErrors` and add `pristine` classname in the `NxTextInput` component, for this it assertion, should we just test if the input does not have the `pristine` classname? Please kindly let me know your opinions, thanks.